From a3c03d41812f9c8deffda4e46d6a4b35e038de60 Mon Sep 17 00:00:00 2001
From: JunSung Choi <junsungc@gooroom.kr>
Date: Fri, 24 May 2019 20:38:08 +0900
Subject: [PATCH 17/63] Add and show gooroom protocol logo with short brand
 text and fake scheme gooroom:// instead of chrome:// in omnibox. Add
 'GooroomURLRewrite::ChangeGooroomScheme' static function

---
 chrome/app/chromium_strings.grd                    | 11 +++++
 chrome/browser/chrome_content_browser_client.cc    |  8 ++++
 .../toolbar/chrome_location_bar_model_delegate.cc  | 14 +++++++
 .../ui/views/location_bar/location_icon_view.cc    | 11 +++++
 .../ui/views/page_info/page_info_bubble_view.cc    |  6 +++
 components/components_chromium_strings.grd         |  2 +-
 components/omnibox/browser/BUILD.gn                |  3 ++
 .../browser/vector_icons/gooroom_product.icon      | 47 ++++++++++++++++++++++
 .../browser/vector_icons/gooroom_product_1x.icon   | 47 ++++++++++++++++++++++
 .../strings/components_chromium_strings_ko.xtb     |  4 +-
 gooroom/URLRewrite/gooroom_url_rewrite.cc          | 10 +++++
 gooroom/URLRewrite/gooroom_url_rewrite.h           |  2 +
 12 files changed, 162 insertions(+), 3 deletions(-)
 create mode 100644 components/omnibox/browser/vector_icons/gooroom_product.icon
 create mode 100644 components/omnibox/browser/vector_icons/gooroom_product_1x.icon

diff --git a/chrome/app/chromium_strings.grd b/chrome/app/chromium_strings.grd
index dc34de77c..750da9d65 100644
--- a/chrome/app/chromium_strings.grd
+++ b/chrome/app/chromium_strings.grd
@@ -129,12 +129,23 @@ If you update this file, be sure also to update google_chrome_strings.grd. -->
       <message name="IDS_PROFILES_DISCONNECT_MANAGED_PROFILE_TEXT" desc="Message explaining to the user what will happen if they disconnect the managed profile.">
         Disconnecting <ph name="USERNAME">$1<ex>someone@example.com</ex></ph> will clear your history, bookmarks, settings, and other Chromium data stored on this device. Data stored in your Google Account will not be cleared and can be managed on <ph name="GOOGLE_DASHBOARD_LINK">&lt;a target="_blank" href="$2"&gt;</ph>Google Dashboard<ph name="END_GOOGLE_DASHBOARD_LINK">&lt;/a&gt;</ph>.
       </message>
+      <if expr="_gooroom">
+        <!-- String "Gooroom" immediately after Gooroom protocol logo -->
+        <message name="IDS_PRODUCT_NAME" desc="The Gooroom Browser application name">
+          Gooroom Browser
+        </message>
+        <message name="IDS_SHORT_PRODUCT_NAME" desc="The Gooroom Browser application short name.">
+          Gooroom
+        </message>
+      </if>
+      <if expr="not _gooroom">
       <message name="IDS_PRODUCT_NAME" desc="The Chrome application name">
         Chromium
       </message>
       <message name="IDS_SHORT_PRODUCT_NAME" desc="The Chrome application short name.">
         Chromium
       </message>
+      </if>
       <if expr="is_win">
         <message name="IDS_SXS_SHORTCUT_NAME" desc="Unused in Chromium builds" translateable="false">
         </message>
diff --git a/chrome/browser/chrome_content_browser_client.cc b/chrome/browser/chrome_content_browser_client.cc
index dfdf3263e..4980bd894 100644
--- a/chrome/browser/chrome_content_browser_client.cc
+++ b/chrome/browser/chrome_content_browser_client.cc
@@ -593,6 +593,10 @@
 #include "services/content/simple_browser/public/mojom/constants.mojom.h"
 #endif
 
+#if BUILDFLAG(ENABLE_GOOROOM)
+#include "gooroom/URLRewrite/gooroom_url_rewrite.h"
+#endif
+
 #if BUILDFLAG(ENABLE_ISOLATED_XR_SERVICE)
 #include "device/vr/public/mojom/isolated_xr_service.mojom.h"
 #endif
@@ -1073,6 +1077,10 @@ ChromeContentBrowserClient::ChromeContentBrowserClient(
   extra_parts_.push_back(new ChromeContentBrowserClientExtensionsPart);
 #endif
 
+#if BUILDFLAG(ENABLE_GOOROOM)
+  extra_parts_.push_back(new GooroomURLRewrite);
+#endif
+
   gpu_binder_registry_.AddInterface(
       base::Bind(&metrics::CallStackProfileCollector::Create));
 }
diff --git a/chrome/browser/ui/toolbar/chrome_location_bar_model_delegate.cc b/chrome/browser/ui/toolbar/chrome_location_bar_model_delegate.cc
index 75c4de647..7c71f9362 100644
--- a/chrome/browser/ui/toolbar/chrome_location_bar_model_delegate.cc
+++ b/chrome/browser/ui/toolbar/chrome_location_bar_model_delegate.cc
@@ -4,6 +4,9 @@
 
 #include "chrome/browser/ui/toolbar/chrome_location_bar_model_delegate.h"
 
+#if defined(GOOROOM_BUILD)
+#include "gooroom/URLRewrite/gooroom_url_rewrite.h"
+#endif
 #include "base/logging.h"
 #include "build/build_config.h"
 #include "chrome/browser/autocomplete/autocomplete_classifier_factory.h"
@@ -67,6 +70,11 @@ bool ChromeLocationBarModelDelegate::GetURL(GURL* url) const {
     return false;
 
   *url = ShouldDisplayURL() ? entry->GetVirtualURL() : GURL();
+#if BUILDFLAG(ENABLE_GOOROOM)
+  // Change "chrome://" to "gooroom://"
+  *url = GooroomURLRewrite::ChangeGooroomScheme(*url);
+#endif
+
   return true;
 }
 
@@ -143,6 +151,12 @@ const gfx::VectorIcon* ChromeLocationBarModelDelegate::GetVectorIconOverride()
 
   if (url.SchemeIs(extensions::kExtensionScheme))
     return &omnibox::kExtensionAppIcon;
+
+#if defined(GOOROOM_BUILD)
+  // Show gooroom protocol logo prior to url "chrome://*".
+  if (url.SchemeIs(content::kGooroomUIScheme))
+    return &omnibox::kGooroomProductIcon;
+#endif // GOOROOM_BUILD
 #endif
 
   return nullptr;
diff --git a/chrome/browser/ui/views/location_bar/location_icon_view.cc b/chrome/browser/ui/views/location_bar/location_icon_view.cc
index 30caa57a5..3d68608e0 100644
--- a/chrome/browser/ui/views/location_bar/location_icon_view.cc
+++ b/chrome/browser/ui/views/location_bar/location_icon_view.cc
@@ -120,6 +120,10 @@ bool LocationIconView::ShouldShowText() const {
     const GURL& url = location_bar_model->GetURL();
     if (url.SchemeIs(content::kChromeUIScheme) ||
         url.SchemeIs(extensions::kExtensionScheme) ||
+#if defined(GOOROOM_BUILD)
+      // Show a string "Gooroom" immediately after Gooroom protocol logo
+        url.SchemeIs(content::kGooroomUIScheme) ||
+#endif
         url.SchemeIs(url::kFileScheme))
       return true;
   }
@@ -132,9 +136,16 @@ const views::InkDrop* LocationIconView::get_ink_drop_for_testing() {
 }
 
 base::string16 LocationIconView::GetText() const {
+#if defined(GOOROOM_BUILD)
+  // Show a string "안전한 Gooroom 페이지를..." after clicking Gooroom protocol logo
+  if (delegate_->GetLocationBarModel()->GetURL().SchemeIs(content::kGooroomUIScheme)) {
+    return l10n_util::GetStringUTF16(IDS_SHORT_PRODUCT_NAME);
+  }
+#else
   if (delegate_->GetLocationBarModel()->GetURL().SchemeIs(
           content::kChromeUIScheme))
     return l10n_util::GetStringUTF16(IDS_SHORT_PRODUCT_NAME);
+#endif
 
   if (delegate_->GetLocationBarModel()->GetURL().SchemeIs(url::kFileScheme))
     return l10n_util::GetStringUTF16(IDS_OMNIBOX_FILE);
diff --git a/chrome/browser/ui/views/page_info/page_info_bubble_view.cc b/chrome/browser/ui/views/page_info/page_info_bubble_view.cc
index ebe5987d5..270543cf3 100644
--- a/chrome/browser/ui/views/page_info/page_info_bubble_view.cc
+++ b/chrome/browser/ui/views/page_info/page_info_bubble_view.cc
@@ -386,6 +386,9 @@ InternalPageInfoBubbleView::InternalPageInfoBubbleView(
   } else if (url.SchemeIs(url::kFileScheme)) {
     text = IDS_PAGE_INFO_FILE_PAGE;
   } else if (!url.SchemeIs(content::kChromeUIScheme) &&
+#if defined(GOOROOM_BUILD)
+             !url.SchemeIs(content::kGooroomUIScheme) &&
+#endif
              !url.SchemeIs(content::kChromeDevToolsScheme)) {
     NOTREACHED();
   }
@@ -430,6 +433,9 @@ views::BubbleDialogDelegateView* PageInfoBubbleView::CreatePageInfoBubble(
   gfx::NativeView parent_view = platform_util::GetViewForWindow(parent_window);
 
   if (url.SchemeIs(content::kChromeUIScheme) ||
+#if defined(GOOROOM_BUILD)
+      url.SchemeIs(content::kGooroomUIScheme) ||
+#endif
       url.SchemeIs(content::kChromeDevToolsScheme) ||
       url.SchemeIs(extensions::kExtensionScheme) ||
       url.SchemeIs(content::kViewSourceScheme) ||
diff --git a/components/components_chromium_strings.grd b/components/components_chromium_strings.grd
index cb8023831..94ba03de7 100644
--- a/components/components_chromium_strings.grd
+++ b/components/components_chromium_strings.grd
@@ -210,7 +210,7 @@
 
       <!-- Page Info -->
       <message name="IDS_PAGE_INFO_INTERNAL_PAGE" desc="Message to display in the page info bubble when the page you are on is a chrome:// page or about:something.">
-        You're viewing a secure Chromium page
+        You're viewing a secure Gooroom page
       </message>
       <message name="IDS_PAGE_INFO_SECURITY_TAB_SECURE_IDENTITY_VERIFIED" desc="The text of the identity section when the page is secure and uses a valid certificate">
         Chromium verified that <ph name="ISSUER">$1<ex>VeriSign</ex></ph> issued this website's certificate.
diff --git a/components/omnibox/browser/BUILD.gn b/components/omnibox/browser/BUILD.gn
index a60bf7393..05671e140 100644
--- a/components/omnibox/browser/BUILD.gn
+++ b/components/omnibox/browser/BUILD.gn
@@ -53,6 +53,9 @@ aggregate_vector_icons("omnibox_vector_icons") {
     "star.icon",
     "switch.icon",
     "tab.icon",
+    # convert Gooroom protocol logo, .svg to skia
+    "gooroom_product_1x.icon",
+    "gooroom_product.icon",
   ]
 }
 
diff --git a/components/omnibox/browser/vector_icons/gooroom_product.icon b/components/omnibox/browser/vector_icons/gooroom_product.icon
new file mode 100644
index 000000000..b62188f9d
--- /dev/null
+++ b/components/omnibox/browser/vector_icons/gooroom_product.icon
@@ -0,0 +1,47 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+CANVAS_DIMENSIONS, 16,
+MOVE_TO, 7.85f, 15.79f,
+R_ARC_TO, 6.67f, 6.67f, 0, 0, 1, -0.78f, -0.31f,
+R_CUBIC_TO, -0.54f, -0.27f, -1.18f, -0.72f, -1.74f, -1.23f,
+R_ARC_TO, 18.02f, 18.02f, 0, 0, 1, -1.01f, -1.02f,
+CUBIC_TO, 3.19f, 11.96f, 2.27f, 10.35f, 1.8f, 8.84f,
+R_CUBIC_TO, -0.29f, -0.92f, -0.36f, -1.48f, -0.34f, -2.55f,
+R_CUBIC_TO, 0.01f, -0.64f, 0.03f, -0.77f, 0.13f, -1.17f,
+R_CUBIC_TO, 0.38f, -1.48f, 1.52f, -2.96f, 2.99f, -3.88f,
+R_CUBIC_TO, 1.93f, -1.21f, 4.26f, -1.38f, 6.33f, -0.48f,
+R_CUBIC_TO, 1.55f, 0.68f, 2.89f, 1.98f, 3.43f, 3.35f,
+R_CUBIC_TO, 0.09f, 0.22f, 0.15f, 0.48f, 0.18f, 0.68f,
+R_CUBIC_TO, 0.02f, 0.15f, 0.01f, 0.17f, -0.03f, 0.13f,
+R_ARC_TO, 4.55f, 4.55f, 0, 0, 0, -0.34f, -0.18f,
+R_CUBIC_TO, -1.19f, -0.59f, -2.91f, -1.18f, -4.37f, -1.5f,
+R_CUBIC_TO, -1.3f, -0.28f, -2.5f, -0.36f, -3.37f, -0.2f,
+R_CUBIC_TO, -0.59f, 0.1f, -1.13f, 0.34f, -1.44f, 0.64f,
+R_LINE_TO, -0.15f, 0.15f,
+R_LINE_TO, -0.07f, 0.3f,
+R_CUBIC_TO, -0.32f, 1.35f, -0.4f, 2.73f, -0.23f, 3.91f,
+R_CUBIC_TO, 0.07f, 0.47f, 0.14f, 0.7f, 0.33f, 1.13f,
+R_CUBIC_TO, 0.39f, 0.86f, 0.9f, 1.62f, 1.56f, 2.32f,
+R_CUBIC_TO, 0.58f, 0.61f, 1.06f, 1.01f, 1.57f, 1.3f,
+R_CUBIC_TO, 0.2f, 0.11f, 0.2f, 0.11f, 0.28f, 0.08f,
+R_CUBIC_TO, 0.47f, -0.17f, 0.9f, -0.47f, 1.45f, -1.02f,
+R_ARC_TO, 6.94f, 6.94f, 0, 0, 0, 1.5f, -2.22f,
+R_CUBIC_TO, 0.1f, -0.24f, 0.22f, -0.59f, 0.22f, -0.65f,
+R_CUBIC_TO, 0, -0.06f, -0.34f, -0.32f, -0.58f, -0.45f,
+R_CUBIC_TO, -0.54f, -0.28f, -1.12f, -0.39f, -1.94f, -0.39f,
+R_CUBIC_TO, -0.24f, 0, -0.49f, 0.01f, -0.54f, 0.02f,
+R_LINE_TO, -0.1f, 0.02f,
+R_LINE_TO, -0.01f, -0.43f,
+R_CUBIC_TO, -0.01f, -0.24f, -0.02f, -0.69f, -0.02f, -1.01f,
+LINE_TO, 8.2f, 6.13f,
+R_LINE_TO, 0.32f, -0.01f,
+R_CUBIC_TO, 0.5f, -0.02f, 1.23f, 0.04f, 1.84f, 0.16f,
+R_ARC_TO, 9.73f, 9.73f, 0, 0, 1, 3.77f, 1.64f,
+R_CUBIC_TO, 0.45f, 0.33f, 0.41f, 0.27f, 0.38f, 0.45f,
+R_CUBIC_TO, -0.11f, 0.6f, -0.4f, 1.58f, -0.66f, 2.18f,
+R_CUBIC_TO, -0.81f, 1.91f, -2.16f, 3.42f, -4.05f, 4.55f,
+R_CUBIC_TO, -0.42f, 0.25f, -1.19f, 0.63f, -1.36f, 0.68f,
+R_ARC_TO, 1.63f, 1.63f, 0, 0, 1, -0.58f, 0.01f,
+CLOSE
diff --git a/components/omnibox/browser/vector_icons/gooroom_product_1x.icon b/components/omnibox/browser/vector_icons/gooroom_product_1x.icon
new file mode 100644
index 000000000..b62188f9d
--- /dev/null
+++ b/components/omnibox/browser/vector_icons/gooroom_product_1x.icon
@@ -0,0 +1,47 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+CANVAS_DIMENSIONS, 16,
+MOVE_TO, 7.85f, 15.79f,
+R_ARC_TO, 6.67f, 6.67f, 0, 0, 1, -0.78f, -0.31f,
+R_CUBIC_TO, -0.54f, -0.27f, -1.18f, -0.72f, -1.74f, -1.23f,
+R_ARC_TO, 18.02f, 18.02f, 0, 0, 1, -1.01f, -1.02f,
+CUBIC_TO, 3.19f, 11.96f, 2.27f, 10.35f, 1.8f, 8.84f,
+R_CUBIC_TO, -0.29f, -0.92f, -0.36f, -1.48f, -0.34f, -2.55f,
+R_CUBIC_TO, 0.01f, -0.64f, 0.03f, -0.77f, 0.13f, -1.17f,
+R_CUBIC_TO, 0.38f, -1.48f, 1.52f, -2.96f, 2.99f, -3.88f,
+R_CUBIC_TO, 1.93f, -1.21f, 4.26f, -1.38f, 6.33f, -0.48f,
+R_CUBIC_TO, 1.55f, 0.68f, 2.89f, 1.98f, 3.43f, 3.35f,
+R_CUBIC_TO, 0.09f, 0.22f, 0.15f, 0.48f, 0.18f, 0.68f,
+R_CUBIC_TO, 0.02f, 0.15f, 0.01f, 0.17f, -0.03f, 0.13f,
+R_ARC_TO, 4.55f, 4.55f, 0, 0, 0, -0.34f, -0.18f,
+R_CUBIC_TO, -1.19f, -0.59f, -2.91f, -1.18f, -4.37f, -1.5f,
+R_CUBIC_TO, -1.3f, -0.28f, -2.5f, -0.36f, -3.37f, -0.2f,
+R_CUBIC_TO, -0.59f, 0.1f, -1.13f, 0.34f, -1.44f, 0.64f,
+R_LINE_TO, -0.15f, 0.15f,
+R_LINE_TO, -0.07f, 0.3f,
+R_CUBIC_TO, -0.32f, 1.35f, -0.4f, 2.73f, -0.23f, 3.91f,
+R_CUBIC_TO, 0.07f, 0.47f, 0.14f, 0.7f, 0.33f, 1.13f,
+R_CUBIC_TO, 0.39f, 0.86f, 0.9f, 1.62f, 1.56f, 2.32f,
+R_CUBIC_TO, 0.58f, 0.61f, 1.06f, 1.01f, 1.57f, 1.3f,
+R_CUBIC_TO, 0.2f, 0.11f, 0.2f, 0.11f, 0.28f, 0.08f,
+R_CUBIC_TO, 0.47f, -0.17f, 0.9f, -0.47f, 1.45f, -1.02f,
+R_ARC_TO, 6.94f, 6.94f, 0, 0, 0, 1.5f, -2.22f,
+R_CUBIC_TO, 0.1f, -0.24f, 0.22f, -0.59f, 0.22f, -0.65f,
+R_CUBIC_TO, 0, -0.06f, -0.34f, -0.32f, -0.58f, -0.45f,
+R_CUBIC_TO, -0.54f, -0.28f, -1.12f, -0.39f, -1.94f, -0.39f,
+R_CUBIC_TO, -0.24f, 0, -0.49f, 0.01f, -0.54f, 0.02f,
+R_LINE_TO, -0.1f, 0.02f,
+R_LINE_TO, -0.01f, -0.43f,
+R_CUBIC_TO, -0.01f, -0.24f, -0.02f, -0.69f, -0.02f, -1.01f,
+LINE_TO, 8.2f, 6.13f,
+R_LINE_TO, 0.32f, -0.01f,
+R_CUBIC_TO, 0.5f, -0.02f, 1.23f, 0.04f, 1.84f, 0.16f,
+R_ARC_TO, 9.73f, 9.73f, 0, 0, 1, 3.77f, 1.64f,
+R_CUBIC_TO, 0.45f, 0.33f, 0.41f, 0.27f, 0.38f, 0.45f,
+R_CUBIC_TO, -0.11f, 0.6f, -0.4f, 1.58f, -0.66f, 2.18f,
+R_CUBIC_TO, -0.81f, 1.91f, -2.16f, 3.42f, -4.05f, 4.55f,
+R_CUBIC_TO, -0.42f, 0.25f, -1.19f, 0.63f, -1.36f, 0.68f,
+R_ARC_TO, 1.63f, 1.63f, 0, 0, 1, -0.58f, 0.01f,
+CLOSE
diff --git a/components/strings/components_chromium_strings_ko.xtb b/components/strings/components_chromium_strings_ko.xtb
index d30ef37a8..808ed89b1 100644
--- a/components/strings/components_chromium_strings_ko.xtb
+++ b/components/strings/components_chromium_strings_ko.xtb
@@ -4,7 +4,7 @@
 <translation id="130631256467250065">기기를 다시 시작할 때 변경사항이 적용됩니다.</translation>
 <translation id="1838412507805038478">Chromium이 <ph name="ISSUER" />에서 이 웹사이트의 인증서를 발행했음을 확인했습니다.</translation>
 <translation id="275588974610408078">비정상 종료 보고서는 Chromium에서 사용할 수 없습니다.</translation>
-<translation id="3064346599913645280">안전한 Chromium 페이지를 보는 중</translation>
+<translation id="32013959327000922">안전한 Gooroom 페이지를 보는 중</translation>
 <translation id="3550966579244642892">Chromium OS 초기 설정이 완료되지 않았습니다.</translation>
 <translation id="4365115785552740256">Chromium은 <ph name="BEGIN_LINK_CHROMIUM" />Chromium<ph name="END_LINK_CHROMIUM" /> 오픈소스 프로젝트를 비롯한 여러 <ph name="BEGIN_LINK_OSS" />오픈소스 소프트웨어<ph name="END_LINK_OSS" />에 기초해 만들어진 브라우저입니다.</translation>
 <translation id="4559775032954821361">Chromium 메뉴 &gt;
@@ -36,4 +36,4 @@
 <translation id="8187289872471304532">애플리케이션 &gt; 시스템 환경설정 &gt; 네트워크 &gt; 고급 &gt; 프록시로 이동한 다음
           선택된 프록시를 선택 취소합니다.</translation>
 <translation id="8684913864886094367">Chromium이 제대로 종료되지 않았습니다.</translation>
-</translationbundle>
\ No newline at end of file
+</translationbundle>
diff --git a/gooroom/URLRewrite/gooroom_url_rewrite.cc b/gooroom/URLRewrite/gooroom_url_rewrite.cc
index ad2ccc8cb..8f48b301e 100644
--- a/gooroom/URLRewrite/gooroom_url_rewrite.cc
+++ b/gooroom/URLRewrite/gooroom_url_rewrite.cc
@@ -9,6 +9,16 @@
 #include "content/public/common/web_preferences.h"
 #include "ui/base/l10n/l10n_util.h"
 
+GURL GooroomURLRewrite::ChangeGooroomScheme(GURL url){
+  // Change "chrome://" to "gooroom://"
+  if(url.scheme() == "chrome"){
+    std::string new_url = "gooroom://" + url.GetContent();
+    GURL changed_url = GURL(new_url);
+    return changed_url;
+  }
+  return url;
+}
+
 bool HandleGooroomURLRewrite(GURL* url,
                              content::BrowserContext* browser_context) {
   // change "gooroom://*" to "chrome://*"
diff --git a/gooroom/URLRewrite/gooroom_url_rewrite.h b/gooroom/URLRewrite/gooroom_url_rewrite.h
index e05a82d3e..682909b66 100644
--- a/gooroom/URLRewrite/gooroom_url_rewrite.h
+++ b/gooroom/URLRewrite/gooroom_url_rewrite.h
@@ -28,6 +28,8 @@ class GooroomURLRewrite : public ChromeContentBrowserClientParts {
  public:
   ~GooroomURLRewrite() override {}
 
+  static GURL ChangeGooroomScheme(GURL url);
+
   void BrowserURLHandlerCreated(content::BrowserURLHandler* handler) override;
   void OverrideWebkitPrefs(content::RenderViewHost* rvh,
                            content::WebPreferences* web_prefs) override;
-- 
2.11.0

